src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -370,87 +383,167 @@ function addHeadAndTrailingLogToBreach({ result, sensor, database }) {
+      temperature: Math.max(...sensorLogGroup.map(({ max }) => max)),
+    }),
+    createGenericSensorLog({
+      ...medianSensorLog,
+      aggregation: SENSOR_LOG_FULL_AGGREGATE_TYPE,
+      temperature: Math.min(...sensorLogGroup.map(({ min }) => min)),
+    }),
+  ];
+}
 
-  const logsToDelete = sensorLogs
-    .filtered('aggregation == "preAggregate" && timestamp < $0', lastAggregationTo)
-    .slice();
-  const fullAggregationDeletions = logsToDelete.length;
-  database.write(() => {
-    logsToDelete.forEach(sensorLog => {
-      database.delete('SensorLog', sensorLog);
-    });
+/**
+ * Utility method used to aggregate pre-aggregate sensor logs for a given
+ * sensor.
+ *
+ * Pre-aggregate sensor logs: sensor log for a 20 minute time interval.
+ * Aggregate sensor log: aggregated sensor log for a 8 hour interval.
+ *
+ * Takes groups of pre-aggregate sensor logs and creates two new
+ * sensor logs for each group - each having a temperature equal
+ * to the max and min temperature within the group of sensor logs.
+ * The timestamp is equal to the median timestamp of all sensor
+ * logs within the group.
+ *
+ * Each group to be aggregated is a collection of sequential sensor logs which fit
+ * within an 8 hour interval beginning with the first sensor logs timestamp. Each
+ * pre-aggregate sensor log is for a 20 minute interval, so there will be a
+ * maximum of 24 sensor logs for each full aggregate log.
+ * If within the 8 hour period a breach occurred, there will be less sensorlogs,
+ * to a minimum of 1.
+ * @param {object}       result   object containing the result of aggregation
+ * @param {Realm.Sensor} sensor   The Sensor object for which the logs should be aggregated
+ * @param {Realm}        database App-wide database entry point
+ */
+function doFullAggregation({ result, sensor, database }) {
+  const { sensorLogs } = sensor;
+  if (!sensorLogs || sensorLogs.length === 0) return null;
+  // Want to leave at least the last 32 hours unaggregated. Also want to only
+  // aggregate the most recently added preAggregate logs, from the most
+  // most recent fullAggregation log.
+  const mostRecentFullAggregateTimestamp = sensorLogs
+    .filtered('aggregation == $0', SENSOR_LOG_FULL_AGGREGATE_TYPE)
+    .max('timestamp');
+  const mostRecentPreAggregateTimestamp = sensorLogs
+    .filtered('aggregation == $0', SENSOR_LOG_PRE_AGGREGATE_TYPE)
+    .max('timestamp');
+  const maxTimestampForDeletion = new Date().getTime() - SENSOR_LOG_NO_DELETION_INTERVAL;
+  // Starting timestamp is either from the first fullAggregate timestamp, or beginning of time
+  const aggregationIntervalStart = (mostRecentFullAggregateTimestamp || -Infinity) + 1;
+  // Ending timestamp is either from the most recent preAggregate or from now, less the
+  // no full aggregate period (32 hours)
+  const aggregationIntervalEnd =
+    (mostRecentPreAggregateTimestamp || new Date().getTime()) - NO_FULL_AGGREGATE_PERIOD;
```

Comment:
i think this should just be maxTimestampForDeletion ?

Reply:
7fb11aeda7c1f2c9c78024d9f123eb6166993196 - I think using the date of the last preaggregate is right - if we have large intervals between syncing it could incorrectly aggregate

Reply:
true

Reply:
actually, full aggregation would happen after a successful sync, at this stage we should have all of the info pulled from the sensor, so either way would work. i like the pre-emptive thinking though